### PR TITLE
FOGL-2181: Add filters to system test

### PR DESCRIPTION
**Pull request by Singhal-Vaibhav**
Playback plugin with multiple filters in pipeline - delta filter, RMS filter, rate filter, scale filter, asset filter and meta filter and send data to PI

### CODE REVIEW

File: tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+        yield self.start_south_north
+
+        remove_directories("/tmp/foglamp-south-{}".format(south_plugin))
+        remove_directories("/tmp/foglamp-filter-{}".format("scale"))
+        remove_directories("/tmp/foglamp-filter-{}".format("asset"))
+        remove_directories("/tmp/foglamp-filter-{}".format("rate"))
+        remove_directories("/tmp/foglamp-filter-{}".format("delta"))
+        remove_directories("/tmp/foglamp-filter-{}".format("rms"))
+        remove_directories("/tmp/foglamp-filter-{}".format("metadata"))
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py]: you can save some lines with:

```
filters = ["scale", "asset", "rate", "delta", "rms", "metadata"]
for f in filters:
    remove_directories("/tmp/foglamp-filter-{}".format(f)
```

File: tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        f.write(CSV_HEADERS)
+        for _items in CSV_DATA.split(","):
+            f.write("\n{}".format(_items))
+        f.close()
+
+        south_plugin = "playback"
+        add_south(south_plugin, south_branch, foglamp_url, service_name=SVC_NAME,
+                  config=south_config, start_service=False)
+
+        filter_cfg = {"enable": "true"}
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py]: filter_cfg_scale

File: tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                                                  "action": "rename",
+                                                  "asset_name": asset_name}]},
+                            "enable": "true"}
+        add_filter("asset", filter_branch, "fasset", filter_cfg_asset, foglamp_url, SVC_NAME)
+
+        # I/P 1000, 2000, 2100, 4000 -> O/P 2000, 2100, 4000
+        filter_cfg_rate = {"trigger": "ivalue > 1200", "untrigger": "ivalue < 1100", "preTrigger": "0", "enable": "true"}
+        add_filter("rate", filter_branch, "frate", filter_cfg_rate, foglamp_url, SVC_NAME)
+
+        # I/P 1000, 2000, 2100, 40000 -> O/P 2000, 4000
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py]: can you please elaborate more here? for delta tolerance logic for O/P

File: tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        filter_cfg_delta = {"tolerance": "20", "enable": "true"}
+        add_filter("delta", filter_branch, "fdelta", filter_cfg_delta , foglamp_url, SVC_NAME)
+
+        # I/P 2000, 4000 -> O/P rms=3162.2776601684, rms_peak=2000
+        filter_cfg_rms = {"assetName": "%a_RMS", "samples": "2", "peak": "true", "enable": "true"}
+        add_filter("rms", filter_branch, "frms", filter_cfg_rms, foglamp_url, SVC_NAME)
+
+        add_filter("metadata", filter_branch, "fmeta", filter_cfg, foglamp_url, SVC_NAME)
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_csv_multi_filter_pi.py]: even if just a re definition `filter_cfg_metadata = {"enable": "true"}` please define

File: tests/system/python/conftest.py [hunk 5]
```diff
@@ -238,9 +238,9 @@ def _add_filter(filter_plugin, filter_plugin_branch, filter_name, filter_config,
         r = conn.getresponse()
         assert 200 == r.status
         res = r.read().decode()
-        expected = "Filter pipeline {{'pipeline': ['{}']}} updated successfully".format(filter_name)
         jdoc = json.loads(res)
-        assert expected == jdoc["result"]
+        # Asset newly added filter exist in request's response
+        assert filter_name in jdoc["result"]
```
[the remark this answers is not in the document]
Comment (Singhal-Vaibhav) [reply tests/system/python/conftest.py]: This was true for a fresh system and with single filter, if we are adding multiple filters, we get a pipeline of newly added filter + old added filters. If we want to have an exact assertion then we have to make an additional GET request. Do we want that? This approach just verifies that newly added filter exists in json response of PUT request

**Pull request merged**